components/Repo/Add.js

```diff
@@ -55,44 +81,143 @@ const styles = {
   })
 }
 
+const TemplatePicker = compose(
+  withT,
+  withRouter,
+  graphql(getTemplateRepos, {
+    options: () => ({
+      fetchPolicy: 'network-only'
+    }),
+    skip: ({ isTemplate }) => isTemplate
+  })
+)(({ t, data, schema, onChange, isTemplate }) => {
+  const [schemaOptions] = useState(
+    schemaKeys.map(key => ({
+      value: key,
+      text: t(`repo/add/template/${key}`, null, key)
+    }))
+  )
+  const [templateOptions, setTemplateOptions] = useState(schemaOptions)
+  const [templateFilter, setTemplateFilter] = useState('')
+  const [template, setTemplate] = useState({
+    value: schema,
+    text: t(`repo/add/template/${schema}`, null, schema)
+  })
+
+  useEffect(() => {
+    setTemplateOptions(
```

Comment:
If an effect always sets a state you should use `useMemo` instead of `useState` and `useEffect`:
```js
const templateOptions = useMemo(mapAndFilterStuff, [data, templateFilter])
```